src/shared/components/query/QueryAndDownloadTab.spec.tsx

```diff
@@ -1,18 +1,107 @@
         assert.equal(comp.find(Tab).length, 1);
     });
 
+    describe('VirtualStudyOperations', () => {
+        let store:QueryStore
+        let virtualStudies: VirtualStudy[]
+        let component : ReactWrapper<any, any>
+        let deleteVirtualStudyStub: sinon.SinonStub
+        let addVirtualStudyStub: sinon.SinonStub
+
+
+        before(()=>{
+            
+            virtualStudies = [
+                {
+                "id": "5a6a00b3d4c6b64d64a094a6",
+                "data": {
+                    "name": "1",
+                    "description": "ACC (TCGA): 92 samples\n",
+                    "studies": [
+                    {
+                        "id": "acc_tcga",
+                        "samples": [
+                        "TCGA-PK-A5H9-01",
+                        "TCGA-OR-A5J3-01",
+                        "TCGA-OR-A5JJ-01"
+                        ]
+                    }
+                    ],
+                    "filters": {
+                    "patients": {},
+                    "samples": {}
+                    },
+                    "origin": [
+                    "acc_tcga"
+                    ]
+                } as VirtualStudyData
+                }
+            ]
+
+            sinon.stub(sessionServiceClient, "getUserVirtualStudies").callsFake(function fakeFn() {
+                return new Promise((resolve, reject) => {
+                    resolve(virtualStudies) 
+                });
+            });
+
+             deleteVirtualStudyStub = sinon.stub(sessionServiceClient, "deleteVirtualStudy").callsFake(function fakeFn(id:string) {
+                return new Promise((resolve, reject) => {
+                    //manually update deletedVirtualStudies, since this view component doesn't update during testing
+                    store.deletedVirtualStudies = [id]
+                    resolve() 
+                });
+            });
+
+             addVirtualStudyStub = sinon.stub(sessionServiceClient, "addVirtualStudy").callsFake(function fakeFn(id:string) {
+                return new Promise((resolve, reject) => {
+                    //manually update deletedVirtualStudies, since this view component doesn't update during testing
+                    store.deletedVirtualStudies = []
+                    resolve() 
+                });
+            });
+
+            store = new QueryStore();
+            component = mount(<QueryAndDownloadTabs store={store} />);
+            
+        });
+
+        it("View VirtualStudies", ()=>{
+            assert.deepEqual(
+               Object.keys(store.selectableStudiesSet).length,
+               component.find('[data-test="StudySelect"]').getNodes().length);
+            assert.equal(component.find('[data-test="StudySelect"]').find('.fa-trash').getNodes().length, virtualStudies.length);
+        });
+
+         it("Delete VirtualStudy", ()=>{
+          let button2 = component.find('.fa-trash').at(0)
```

Comment:
@kalletlak we usually add `data-test="virtualStudyItem"` attributes but this works as well. Could just lead to some strange errors if we end up using `.fa-trash` at multiple locations

Reply:
Sure! will update this